skimage/feature/corner.py

```diff
@@ -394,15 +388,20 @@ def _symmetric_compute_eigenvalues(S_elems):
         ith-largest eigenvalue at position (j, k).
     """
 
-    if len(S_elems) == 3:  # Use fast Cython code for 2D
-        eigs = np.stack(_image_orthogonal_matrix22_eigvals(*S_elems))
+    if len(S_elems) == 3:  # Fast explicit formulas for 2D.
+        M00, M01, M11 = S_elems
+        mean = (M00 + M11) / 2
+        hsqrtdet = np.sqrt(M01 ** 2 + ((M00 - M11) / 2) ** 2)
+        eigs = np.empty((2, *M00.shape))  # Preallocating is quite faster.
```

Comment:
This should take the dtype of the input arrays (eg float32 if input was float32) to make the [tests pass again](https://github.com/scikit-image/scikit-image/runs/7356649977?check_suite_focus=true#step:5:672)